lib/src/main/scala/spinal/lib/tools/HDElkDiagramGen.scala

```diff
@@ -0,0 +1,506 @@
+  private def drawClockDomains(): Unit = {
+    builder ++= s"""{id:"ClockDomains",\nchildren:[\n"""
+    for (element <- clkNamesMap) {
+      builder ++= s"""{id:"${element._1}",highlight:${element._2}},\n"""
+    }
+    builder ++= s"""]\n}\n"""
+  }
+
+  /** Integrating all methods to draw an image */
+  def beginDraw(): StringBuilder = {
+    builder ++=
+      s"""
+         |<div id="${topNode.labelName}"></div>
+         |<h3>${topNode.labelName}</h3><br><br><br><br>
+         |<script type="text/javascript">
+         |
+         |var mygraph = {
+         |children:[
+         |""".stripMargin
+    GenColorMap()
+    GenAllNodes()
+    GenAllEdges()
+    drawNodes(topNode)
+    if (clkMap.nonEmpty)
+      drawClockDomains()
+    builder ++= s"""],\n}\nhdelk.layout( mygraph,"${topNode.labelName}");\n</script>\n"""
+    builder
+  }
+}
+
+object HDElkDiagramGen {
+  /** Generating all diagrams */
+  def apply[T <: Component](rtl:SpinalReport[T]){
+    val builder = new StringBuilder()
+    builder ++=
+      s"""<!DOCTYPE html>
+         |<html>
+         |<head>
+         |    <meta charset="UTF-8">
+         |    <title>RTL diagrams based on SpinalHDL code</title>
+         |    <style>
+         |.buttons-container {
+         | display: flex;
+         | justify-content: center;
+         | margin-top:30px;
+         |}
+         |
+         |.buttons-container button {
+         | background-color: #4CAF50;
+         | color: white;
+         | border: none;
+         | padding:10px20px;
+         | margin:010px;
+         | cursor: pointer;
+         | border-radius:5px;
+         |}
+         |
+         |.center-title {
+         |  text-align: center;
+         |  font-size:24px;
+         |  font-weight: bold;
+         |  margin-top:20px;
+         | margin-bottom: 20px;
+         |}
+         |.goTop {
+         |	width: 50px;
+         |	height: 50px;
+         |	background-color: aquamarine;
+         |	font-size: 20px;
+         |	text-align: center;
+         |	line-height: 25px;
+         |	color: azure;
+         |	position: fixed;
+         |	bottom: 50px;
+         |	right: 50px;
+         |	display: none;
+         |}
+         |    </style>
+         |</head>
+         |<body>
+         |<button class="goTop" id="gotop">Go Top</button>
+         |<script>
+         |	var goTop=document.getElementById("gotop")
+         |	    window.onscroll=function(){
+         |		var jhlheight=document.documentElement.scrollTop||document.body.scrollTop
+         |		if(jhlheight>=300){
+         |		goTop.style.display="block"
+         |		}else{
+         |		goTop.style.display="none"
+         |   }
+         |	}
+         |	  goTop.onclick=function(){
+         |		window.scrollTo({
+         |			top:0,
+         |			behavior:"smooth"
+         |		})
+         |	}
+         |</script>
+         |<script src="https://cdn.jsdelivr.net/gh/davidthings/hdelk@master/js/elk.bundled.js"></script>
```

Comment:
This can be a default URL base, however, for users do not have network access might want to change this.
So use a variable is better here.:)